resources/main.mjs

```diff
@@ -1,10 +1,11 @@
 import { BenchmarkRunner } from "./benchmark-runner.mjs";
 import * as Statistics from "./statistics.mjs";
-import { Suites } from "./tests.mjs";
 import { renderMetricView } from "./metric-ui.mjs";
 import { defaultParams, params } from "./shared/params.mjs";
 import { createDeveloperModeContainer } from "./developer-mode.mjs";
 
+const { dataProvider } = await import("../resources/data-provider.mjs");
```

Comment:
I feel like you use `await` here because you have the top level await in data-provider, but that's not necessary, I'd say to just use a static import.

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await

Although we might want to avoid the top level await and instead have a promise we can wait for in the constructor below, or something like that.